lib/DocumentManager.php

```diff
@@ -76,25 +59,25 @@ public function __construct(
         DocumentInspectorFactoryInterface $inspectorFactory,
         $defaultLocale
     ) {
-        // the event dispatcher should be unique to this document manager
-        // instance.
         $this->eventDispatcher = $eventDispatcher;
-
         // the inspector factory provides a way for users to instantiate
         // their own document inspectors.
         $this->inspectorFactory = $inspectorFactory;
 
-        // the metadata factory is currently intended to be the same for all
-        // document manager instances.
-        $this->metadataFactory = $metadataFactory;
-
-        // the PHPCR session SHOULD be unique to all document managers.
-        $this->session = $session;
-
         // instantiate other objects scoped to this document manager.
-        $this->nodeManager = new NodeManager($session);
-        $this->proxyFactory = new ProxyFactory($this, $lazyProxyFactory, $metadataFactory);
-        $this->registry = new DocumentRegistry($defaultLocale);
+        $nodeManager = new NodeManager($session);
+        $proxyFactory = new ProxyFactory($this, $lazyProxyFactory, $metadataFactory);
+        $documentRegistry = new DocumentRegistry($defaultLocale);
+
+        $this->context = new DocumentManagerContext(
+            $this,
```

Comment:
DocumentManagerContext contains an instance of the document manager which instantiatates it.